lib/zoom_api.rb

```
@@ -0,0 +1,88 @@
+      'Authorization' => "Bearer #{auth_token}",
+    }
+  end
+
+  def get_meeting_info(meeting_id)
+    get("/meetings/#{meeting_id}")
+  end
+
+  def add_registrant(meeting_id, body)
+    post("/meetings/#{meeting_id}/registrants", body)
+  end
+
+  def cancel_registrants(meeting_id, registrant_emails)
+    registrants = registrant_emails.map { |registrant| {'email' => registrant } }
+    path = "/meetings/#{meeting_id}/registrants/status"
+    body = { 'action' => 'cancel', 'registrants' => registrants }
+    put(path, body)
+  end
+
+# TODO: I think this was intended to fix up meetings with incorrect settings, but I don't
+# think it's used. Revisit this and remove if there isn't a way to make things easier to support
+# by running some automatic fix up code like this
+# Task: https://app.asana.com/0/1174274412967132/1200467981732246
+#  def update_meeting_for_registration(meeting_id)
+#    Rails.logger.info("Updating meeting for registration: #{meeting_id}")
+#    url = "#{BASE_URL}/meetings/#{meeting_id}"
+#    data = {
+#      'settings' => {
+#        'approval_type' => 0,
+#        'registration_type' => 2,
+#        'registrants_email_notification' => false,
+#        'registrants_confirmation_email' => false
+#      }
+#    }
+#
+#    put(url, data)
+#  end
+
+private
+
+  def get(path, params = {}, headers = {})
+    response = RestClient.get("#{BASE_URL}#{path}", {params: params}.merge(@global_headers.merge(headers)))
+    extract_response(response)
+  end
+
+  def post(path, body, headers = {})
+    response = RestClient.post("#{BASE_URL}#{path}", body.to_json, @global_headers.merge(headers))
+    extract_response(response)
+  end
+
+  def put(path, body, headers = {})
+    response = RestClient.put("#{BASE_URL}#{path}", body.to_json, @global_headers.merge(headers))
+    extract_response(response)
+  end
+
+  def extract_response(response)
+    (response.body.present? ? JSON.parse(response.body) : response)
```

do upstream errors always return either json or an empty body? if not, and this raises a json parse error instead of returning error body text, this could make troubleshooting harder

---

yeah, the responses are JSON or empty.